hamilton/cli/__main__.py

```diff
@@ -0,0 +1,188 @@
+import logging
+from pathlib import Path
+from pprint import pprint
+from typing import Annotated, Any, Optional
+
+import typer
+from pydantic import BaseModel
+
+from hamilton import driver
+from hamilton.cli import commands
+
+logger = logging.getLogger(__name__)
+
+
+class Response(BaseModel):
+    command: str
+    success: bool
+    message: Any
+
+
+class CliState:
+    verbose: Optional[bool] = None
+    json_out: Optional[bool] = (None,)
+    dr: Optional[driver.Driver] = None
+    dataflow_version: Optional[dict] = None
+
+
+cli = typer.Typer()
+state = CliState()
+
+
+# entrypoint for `hamilton` without command
+@cli.callback()
+def main(
+    verbose: Annotated[bool, typer.Option(help="Output all intermediary commands")] = False,
+    json_out: Annotated[
+        bool, typer.Option(help="Output JSON for programmatic use (e.g., CI)")
+    ] = False,
+):
+    """Hamilton CLI"""
```

Comment:
Why the capital letters everywhere?

Reply:
CLI is an acronym :)

Reply:
Command Line Interface (as opposed to GUI, UI, Terminal User Interface, etc.)